module/zfs/zfs_ioctl.c

```diff
@@ -3258,22 +3311,30 @@ zfs_ioc_clone(const char *fsname, nvlist_t *innvl, nvlist_t *outnvl)
 {
 	int error = 0;
 	nvlist_t *nvprops = NULL;
+	nvlist_t *hidden_args = NULL;
+	dsl_crypto_params_t *dcp = NULL;
 	char *origin_name;
 
 	if (nvlist_lookup_string(innvl, "origin", &origin_name) != 0)
 		return (SET_ERROR(EINVAL));
 	(void) nvlist_lookup_nvlist(innvl, "props", &nvprops);
+	(void) nvlist_lookup_nvlist(innvl, ZPOOL_HIDDEN_ARGS, &hidden_args);
```

Comment:
There is actually one right now that I'm dealing with. The zpool create path causes some problems due to things like the timing of creating the root dataset vs when the spa is fully initialized. This is actually what is causing the buildbot failures right now. I'm working through these issues now, but I'm not sure if there is a clean answer (without as you said just not encrypting the initial dataset boilerplate).